PrimerKlarnaSDK/Classes/PrimerKlarnaViewController.swift

```diff
@@ -0,0 +1,134 @@
+public enum KlarnaPaymentCategory: String {
+    case payNow = "pay_now", payLater = "pay_later", payOverTime = "pay_over_time"
+}
+
+public protocol PrimerKlarnaViewControllerDelegate {
+    func primerKlarnaViewDidLoad()
+    func primerKlarnaPaymentSessionCompleted(authorizationToken: String?, error: PrimerKlarnaError?)
+}
+
+public class PrimerKlarnaViewController: UIViewController {
+    
+    var klarnaPaymentView: KlarnaPaymentView!
+    var delegate: PrimerKlarnaViewControllerDelegate
+    private var paymentCategory: KlarnaPaymentCategory
+    private var clientToken: String
+    private var urlScheme: String?
+    private var klarnaPaymentViewHeightConstraint: NSLayoutConstraint!
+    
+    public init(
+        delegate: PrimerKlarnaViewControllerDelegate,
+        paymentCategory: KlarnaPaymentCategory,
+        clientToken: String,
+        urlScheme: String?)
+    {
+        self.delegate = delegate
+        self.paymentCategory = paymentCategory
+        self.clientToken = clientToken
+        self.urlScheme = urlScheme
+        super.init(nibName: nil, bundle: nil)
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override public func viewDidLoad() {
+        super.viewDidLoad()
+        
+        view.backgroundColor = .white
+        
+        klarnaPaymentView = KlarnaPaymentView(category: self.paymentCategory.rawValue, eventListener: self)
+        view.addSubview(klarnaPaymentView)
+        view.heightAnchor.constraint(equalToConstant: 800).isActive = true
+        
+        // Add as subview
+        klarnaPaymentView.translatesAutoresizingMaskIntoConstraints = false
+        klarnaPaymentView.backgroundColor = .red
+        klarnaPaymentView.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: 20.0).isActive = true
+        klarnaPaymentView.topAnchor.constraint(equalTo: view.topAnchor, constant: 20.0).isActive = true
+        klarnaPaymentView.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: -20.0).isActive = true
+        klarnaPaymentViewHeightConstraint = klarnaPaymentView.heightAnchor.constraint(equalToConstant: 0)
+        klarnaPaymentViewHeightConstraint.isActive = true
+        
+        if let urlScheme = urlScheme, let url = URL(string: urlScheme) {
+            klarnaPaymentView.initialize(clientToken: clientToken, returnUrl: url)
+        } else {
+            klarnaPaymentView.initialize(clientToken: clientToken)
+        }
+    }
+}
+
+extension PrimerKlarnaViewController: KlarnaPaymentEventListener {
+    
+    public func klarnaInitialized(paymentView: KlarnaPaymentView) {
+        klarnaPaymentView.load()
+    }
+    
+    public func klarnaLoaded(paymentView: KlarnaPaymentView) {
+        delegate.primerKlarnaViewDidLoad()
+        klarnaPaymentView.authorize(autoFinalize: true, jsonData: nil)
+    }
+    
+    public func klarnaLoadedPaymentReview(paymentView: KlarnaPaymentView) {
+        
+    }
```

Comment:
did we decide not to forward any event back to the merchant via our PrimerKlarnaSDK?

Reply:
Nope, but all their delegate methods are required. So I do have to implement them.